trading/trading/orders/lib/factory.mli

```diff
@@ -0,0 +1,127 @@
+(** Order factory for creating orders from structured data *)
+
+open Trading_base.Types
+open Types
+
+type order_params = {
+  symbol : symbol;
+  side : side;
+  order_type : order_type;
+  quantity : quantity;
+  time_in_force : time_in_force;
+}
+[@@deriving show, eq]
+(** Order creation parameters *)
+
+val create_order : order_params -> order
+(** Create an order from structured parameters *)
+
+(** Validation result for order parameters *)
```

Comment:
Should this use `base/status`? Then the error handling can be more streamlined / consistent with other part of the code base